pio/ws2812/ws2812_parallel.c

```diff
@@ -300,19 +299,18 @@ int main() {
         int brightness = 0;
         uint current = 0;
         for (int i = 0; i < 1000; ++i) {
-            int n = 64;
-
             current_string_out = string0.data;
-            current_string_4color = false;
-            pattern_table[pat].pat(n, t);
+            current_string_4colour = false;
+            pattern_table[pat].pat(NUM_PIXELS, t);
             current_string_out = string1.data;
-            current_string_4color = true;
-            pattern_table[pat].pat(n, t);
+            current_string_4colour = true;
+            pattern_table[pat].pat(NUM_PIXELS, t);
 
-            transform_strings(strings, count_of(strings), colors, n * 4, brightness);
-            dither_values(colors, states[current], states[current ^ 1], n * 4);
+            transform_strings(strings, count_of(strings), colours, NUM_PIXELS * 4, brightness);
+            dither_values(colours, states[current], states[current ^ 1], NUM_PIXELS * 4);
             sem_acquire_blocking(&reset_delay_complete_sem);
-            output_strings_dma(states[current], n * 4);
+            output_strings_dma(states[current], NUM_PIXELS * 4);
+            sleep_ms(10);
```

Comment:
why have you added a sleep here?

Reply:
Because ws2812.c also had a sleep_ms(10) in the loop. I think the animations are too fast without it.

Reply:
it destroys the point of dithering, tho; you should slow down the animation itself rather than the rendering loop

Reply:
I do not see how I could slow down the animation. The pattern_* animation methods just fill the string, and then the string is rendered. And the animation method is called inside the rendering loop, so slowing down filling of the string has the same effect as adding a delay after the strings are rendered at the end of the loop.

Reply:
ok, so only update the string pattern 1 in N times round the loop

Reply:
Removing the sleep_ms(). I did not find a way to update the string pattern only 1 in N times without messing up the animation or ratio between changes and dithering steps.